examples/texture-arrays/constant.frag

```diff
@@ -1,7 +1,7 @@
 #version 450
 
 layout(location = 0) in vec2 v_TexCoord;
-layout(location = 1) flat in int v_Index;
+layout(location = 1) flat in int v_Index;  // dynamically non-uniform
```

Comment:
why is this dynamically non-uniform?

Reply:
My intuition was that "dynamically uniform" meant that it was the same across the entire drawcall which is confirmed by a careful reading of the glsl spec ([pg. 24](https://www.khronos.org/registry/OpenGL/specs/gl/GLSLangSpec.4.40.pdf)) which says (emphasis mine):

> A fragment-shader expression is dynamically uniform if **all fragments** [note - not all fragments in the primative] evaluating it get the same resulting value.

By this rule, something flat shaded over the primative isn't dynamically uniform.

Reply:
I'll need to look at this deeper. The same section you link to also mentions this:

> This is similarly defined for other shader stages, based on the per-instance data they process.

If VS shader is dynamically uniform on per-instance data, that means it's not uniform across the whole draw call like you suggested... It's more suggestive that "dynamically uniform" means uniform across the wave, and each primitive is scheduled in separate waves, and each VS instance is scheduled across different waves. That would match the spec fragment.

Reply:
As a more concrete piece of proof, glslc won't let you index a descriptor array with this value without GL_EXT_nonuniform qualifier and spirv descriptor indexing extensions on.